client/core/core.go

```diff
@@ -2347,6 +2397,88 @@ func handleRevokeMatchMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) erro
 	return tracker.db.UpdateOrder(metaOrder)
 }
 
+// handleTradeSuspensionMsg is called when a trade suspension notification is received.
+func handleTradeSuspensionMsg(c *Core, dc *dexConnection, msg *msgjson.Message) error {
+	var sp msgjson.TradeSuspension
+	err := msg.Unmarshal(&sp)
+	if err != nil {
+		return fmt.Errorf("trade suspension unmarshal error: %v", err)
+	}
+
+	dc.pendingSuspendsMtx.Lock()
+	defer dc.pendingSuspendsMtx.Unlock()
+
+	// Ensure the provided market exists for the dex.
+	dc.marketMtx.RLock()
+	mkt, ok := dc.marketMap[sp.MarketID]
+	dc.marketMtx.RUnlock()
+	if !ok {
+		return fmt.Errorf("no market found with ID %s", sp.MarketID)
+	}
+
+	// Ensure the market is not already suspended.
+	if mkt.Suspended {
+		return fmt.Errorf("market %s for dex %s is already suspended",
+			sp.MarketID, dc.acct.host)
+	}
+
+	// Remove pending suspends for the market.
+	if sched := dc.pendingSuspends[sp.MarketID]; sched != nil {
+		sched.Stop()
+		delete(dc.pendingSuspends, sp.MarketID)
```

Comment:
No real need to delete here since we are about to assign a new one a couple lines down that will overwrite any existing one, although stopping the timer is good.

Reply:
Actually if `sched.Stop()` returns `false`, that means the timer fired already and the market is being set as suspended presently (an AfterFunc closure is running), which will delete the timer that we are about to add when it gets the `pendingSuspendsMtx` lock after `handleTradeSuspensionMsg` returns .  I think we have to return if `Stop` indicates the timer has fired (the handler lost the race with the previously scheduled suspend).

Maybe also put a TODO that we are too late in stopping the timer and we should request current 'config' as there was a race with a suspension message that might have rescheduled the suspend.